ledger/ledger-api-common/src/test/suite/scala/com/digitalasset/platform/server/api/services/grpc/GrpcCommandSubmissionServiceSpec.scala

```diff
@@ -65,7 +57,55 @@ class GrpcCommandSubmissionServiceSpec
         span.end()
       }
     }
+
+    "propagate submission id" in {
+      val expectedSubmissionId = "explicitSubmissionId"
+      val requestWithSubmissionId = submitRequest.copy(
+        commands =
```

Comment:
You may also simplify this with lenses.

Reply:
0d962b0a5dcc4fbeccafbd48fdaaecc1a69e7553